src/kbd-slide.c

```diff
@@ -0,0 +1,217 @@
+/**
+  @file kbd-slide.c
+  Code concerning state of the slide keyboard.
+
+  This file is part of ke-recv.
+
+  Copyright (C) 2020 Arthur Demchenkov <spinal.by@gmail.com>
+
+  This program is free software; you can redistribute it and/or
+  modify it under the terms of the GNU General Public License
+  version 2 as published by the Free Software Foundation.
+
+  This program is distributed in the hope that it will be useful, but
+  WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+  General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program; if not, write to the Free Software
+  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+  02110-1301 USA
+*/
+
+#include <glib.h>
+#include <fcntl.h>
+#include <string.h>
+#include <unistd.h>
+#include <errno.h>
+#include <libevdev/libevdev.h>
+
+#include "events.h"
+#include "kbd-slide.h"
+
+/* Path to the input device directory */
+#define DEV_INPUT_PATH "/dev/input"
+/* Prefix for event files */
+#define EVENT_FILE_PREFIX "event"
+/* The name of device we are going to monitor */
+#define KBD_SLIDE_DEV_NAME "gpio_keys"
+
+static struct libevdev *kbd_slide_dev = NULL;
+static GIOChannel *kbd_slide_iochan = NULL;
+static guint kbd_slide_src_id = 0;
+
+static gboolean kbd_slide_add_handler(void);
+
+/**
+ * Try to set input device as a keypad slide handler
+ * @param path  the device file path to try
+ * @return  TRUE if device is suitable, FALSE otherwise
+ */
+static gboolean
+kbd_slide_set_input_dev(gchar *path)
+{
+    gint fd, rc;
+    struct libevdev *dev = NULL;
+
+    fd = open(path, O_RDONLY|O_NONBLOCK);
+    if (fd < 0)
+        return FALSE;
+
+    rc = libevdev_new_from_fd(fd, &dev);
+    if (rc < 0) {
+        close(fd);
+        return FALSE;
+    }
+
+    if (strcmp(libevdev_get_name(dev), KBD_SLIDE_DEV_NAME) == 0 &&
```

Comment:
Why do we insist on slide being only gpio? I think you should remove that check and handle all slides no matter gpio or not

Reply:
ignore that, it is fixed in the next version